metricbeat/schema/schema.go

```diff
@@ -71,19 +71,13 @@ func (o Object) HasKey(key string) bool {
 
 // ApplyTo adds the fields extracted from data, converted using the schema, to the
 // event map.
-func (s Schema) ApplyTo(event common.MapStr, data map[string]interface{}) (common.MapStr, *Errors) {
```

Comment:
Type switches are mostly going to solve the problem or helper functions. One problem of the implementation could be that it is called error. Perhaps it should be renamed. In case there are some "optional" errors, error is not nil but should be to support the default error handling `if err != nil`. The information returned is more some meta information about the event. 

I understand the reasoning behind the public interface should return error but one advantage of having `*Errors` returned is that it forces the person implementing the code, to deal with it and understand the logic of it. The person implementing it should make a decision on how to deal with the different types of "Errors". As I can see us adding more logic to it in the future we could rename it to "Meta" or something like that and this object has a method `GetError`. So it would not be directly and error. Just some alternative ideas on how we could deal with it.

Thanks for the blog post. One interesting method I was not aware of is `errors.Wrap` that we should probably use more often in our code.